packages/teleport-shared/src/utils/ast-js-utils.ts

```diff
@@ -111,3 +112,85 @@ export const getTSAnnotationForType = (type: any, t = types) => {
       return t.tsUnknownKeyword()
   }
 }
+
+export const createMethodsObject = (
+  methods: Record<string, UIDLEventHandlerStatement[]>,
+  propDefinitions: Record<string, UIDLPropDefinition>,
+  flavor: 'angular' | 'vue',
+  t = types
+) => {
+  return Object.keys(methods).map((eventKey) => {
+    const astStatements = []
+    methods[eventKey].map((statement) => {
+      const astStatement =
+        statement.type === 'propCall'
+          ? createPropCallStatement(statement, propDefinitions, flavor)
+          : createStateChangeStatement(statement)
+
+      if (astStatement) {
+        astStatements.push(astStatement)
+      }
+    })
+    return flavor === 'angular'
+      ? t.classMethod('method', t.identifier(eventKey), [], t.blockStatement(astStatements))
+      : t.objectMethod('method', t.identifier(eventKey), [], t.blockStatement(astStatements))
+  })
+}
+
+const createStateChangeStatement = (statement: UIDLEventHandlerStatement, t = types) => {
+  const { modifies, newState } = statement
+
+  const rightOperand =
+    newState === '$toggle'
+      ? t.unaryExpression('!', t.memberExpression(t.identifier('this'), t.identifier(modifies)))
+      : convertValueToLiteral(newState)
+
+  return t.expressionStatement(
+    t.assignmentExpression(
+      '=',
+      t.memberExpression(t.identifier('this'), t.identifier(modifies)),
+      rightOperand
+    )
+  )
+}
+
+const createPropCallStatement = (
+  eventHandlerStatement: UIDLEventHandlerStatement,
+  propDefinitions: Record<string, UIDLPropDefinition>,
+  flavor: 'angular' | 'vue',
+  t = types
+) => {
+  const { calls: propFunctionKey, args = [] } = eventHandlerStatement
+
+  if (!propFunctionKey) {
+    console.warn(`No prop definition referenced under the "calls" field`)
+    return null
+  }
+
+  const propDefinition = propDefinitions[propFunctionKey]
+
+  if (!propDefinition) {
+    console.warn(`No prop definition was found for function "${propFunctionKey}"`)
+    return null
+  }
+
+  // In vue it's favorable to use $emit for a specific event than sending the function as a prop
+  const emitter =
+    flavor === 'angular'
+      ? t.expressionStatement(
```

Comment:
can this logic be sent as a parameter of how to handle the prop call / event emitter thing?

Reply:
I moved the prop calls to two separate individual functions in angular-base and vue-base. It looks good that way as we don't need to pass flavor flags anymore.